### PR TITLE
fix: add incentive reward estimation

### PR DESCRIPTION
**Pull request by peterslany**
Adding more actual incentive reward estimation without user's need to transact with the protocol.

### CODE REVIEW

File: src/parachain/loans.ts [hunk 1]
```diff
@@ -675,13 +677,120 @@ export class DefaultLoansAPI implements LoansAPI {
...
+            // Total KINT / INTR tokens awarded per block to suppliers of this market
+            this.api.query.loans.rewardSupplySpeed(underlyingCurrencyId),
+            this.api.query.tokens.totalIssuance(lendTokenId),
+        ]);
+        const lastIndex = Big(marketSupplyState.index.toString());
+        const supplyRewardSpeed = Big(marketSupplySpeed.toString());
+        const totalSupply = Big(totalIssuance.toString());
+
+        const deltaBlocks = currentBlockNumber - marketSupplyState.block.toNumber();
+        const deltaIndex = supplyRewardSpeed.div(totalSupply).mul(deltaBlocks);
```
Comment (daniel-savu) [inline src/parachain/loans.ts:696]: If the `totalSupply` is very large, the division might be rounded to zero. You should first multiply and then divide (in the other function as well)
```suggestion
        const deltaIndex = supplyRewardSpeed.mul(deltaBlocks).div(totalSupply);
```

Comment (peterslany) [reply src/parachain/loans.ts]: Great suggestion, thanks!

File: test/integration/parachain/staging/sequential/loans.test.ts [hunk 2]
```diff
@@ -354,8 +353,45 @@ describe("Loans", () => {
...
+        it("should return correct amount of rewards", async () => {
+            await submitExtrinsic(
+                userInterBtcAPI,
+                await userInterBtcAPI.loans.lend(underlyingCurrency, newMonetaryAmount(1, underlyingCurrency, true)),
+                false
+            );
+
+            const rewards = await userInterBtcAPI.loans.getAccruedRewardsOfAccount(userAccountId);
+
+            expect(rewards.total.toBig().eq(1)).to.be.true;
```
Comment (daniel-savu) [inline test/integration/parachain/staging/sequential/loans.test.ts:394]: You can also lock the collateral and borrow `underlyingCurrencyId2` in the same test, to ensure the borrow reward is correctly calculated as well

Comment (peterslany) [reply test/integration/parachain/staging/sequential/loans.test.ts:394]: Good idea, thanks)

**Pull request merged**